.github/workflows/main.yml

```
@@ -5,23 +5,31 @@ on:
   - pull_request
 jobs:
   test:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-22.04
     strategy:
       fail-fast: false
       matrix:
         python-version:
-          - 3.6
-          - 3.7
-          - 3.8
-          - 3.9
-          - pypy3
+          - "3.8"
+          - "3.9"
+          - "3.10"
+          - "3.11"
+          - "3.12"
+          # - "pypy3.9"
```

It'd be great to keep testing with PyPy. Could you please check if it works with PyPy 3.10?
```suggestion
          - "pypy3.10"
```

---

I'll take a look